lib/util.js

```diff
@@ -155,6 +155,26 @@ export function each (obj, fn, ctx) {
   }
 }
 
+export function fulfill (docs) {
+  if (!Array.isArray(docs)) return docs
+
+  let promised = docs.find(
+    doc => Object.values(doc).find(value => value && !!value.then)
```

Comment:
I like the nonintrusive approach but it does introduce additional computation cost for all other users using vanilla objects. While performance has not really been the biggest focus of this library, this extra iteration on every aggregation run seems significant.

Also a function value may not an object will not survive some of the expression operators such as `set` operations.

It might be better to introduce a new method `runAsPromise()` on the aggregator to make this very explicit if you want direct library support instead of using the `fulfill`  wrapper in your own context.

Unfortunately that would mean using `new mingo.Aggregator(pipeline).runAsPromise(docs)` instead of the short-hand.

Would like to hear your thoughts.